packages/devtools_extensions/lib/src/template/_simulated_devtools_environment/connection_ui/_dtd_connect.dart

```diff
@@ -0,0 +1,136 @@
+// Copyright 2024 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of '../_simulated_devtools_environment.dart';
+
+@visibleForTesting
+class DTDConnectionDisplay extends StatelessWidget {
+  const DTDConnectionDisplay({
+    super.key,
+    required this.simController,
+    required this.connected,
+    required this.onConnectionChange,
+  });
+
+  @visibleForTesting
+  static const totalControlsHeight = 45.0;
+  @visibleForTesting
+  static const totalControlsWidth = 415.0;
+
+  final SimulatedDevToolsController simController;
+  final bool connected;
+  final Future<void> Function(String?) onConnectionChange;
+
+  @override
+  Widget build(BuildContext context) {
+    return _UriConnectionDisplay(
+      simController: simController,
+      connected: connected,
+      connectedLabel: 'DTD connection: ',
+      disconnectedLabel: 'Dart Tooling Daemon Connection',
+      disconnectedHint: '(e.g., ws://127.0.0.1:57624)',
+      // TODO(kenz): this needs handling for bad input.
+      onConnect: (value) async {
+        value = value.trim();
+        if (value.startsWith('127.0.0.1')) {
+          value = 'ws://$value';
+        }
+        await onConnectionChange(value);
+        if (dtdManager.hasConnection) {
+          simController.logInfoEvent(
+            'Connected DTD to ${dtdManager.uri?.toString()}',
+          );
+        } else {
+          simController.logInfoEvent('Failed to connect DTD to $value');
+        }
+      },
+      onDisconnect: () async {
+        await onConnectionChange(null);
+        simController.logInfoEvent('Disconnected from DTD');
+      },
+      currentConnection: () => dtdManager.uri?.toString() ?? '--',
+      help: const DtdHelp(),
+    );
+  }
+}
+
+class DtdHelp extends StatelessWidget {
+  const DtdHelp({super.key});
+
+  static const dtdStartCommand = 'dart tooling-daemon --disable-secrets';
```

Comment:
Yep, will wait to land this PR until the SDK work lands, and then I will also add a note that this requires a minimum SDK version of whatever version we end up needing. Or instead of that, I may just bump the min Dart SDK version of this package to be a version number post DTD. TBD.